docs/design/ConstructiveSolidGeometry.MD

```diff
@@ -0,0 +1,136 @@
+# Constructive Solid Geometry
+
+[Constructive Solid Geometry](https://en.wikipedia.org/wiki/Constructive_solid_geometry) (CSG) allows the modeler to create
+complex shapes using boolean operators to combine simpler
+solid objects.
+
+![](https://upload.wikimedia.org/wikipedia/commons/thumb/8/8b/Csg_tree.png/450px-Csg_tree.png)
+
+In Shards we employ a [Signed Distance Function](https://en.wikipedia.org/wiki/Signed_distance_function) (SDF) to generate implicit 3D shapes such as spheres, cubes, capsules, torii etc. which can be traced by ray-marching through the scene.
+
+We'll explore in depth the binary tree representation, how to flatten it to a list, how the shader consumes it and what's the math behind it.
+
+## The Tree
+
+Each node in the tree is either a simple shape (a leaf node) or an operator that takes the result of one or more child nodes (operands) and combines them. For simplicity we consider unary and binary operators only. Operations on three or more elements can be split into binary operations by nesting some of the elements deeper in the tree. In the example above the right node is an union of three cylinders, but it has been split into two binary union operators.
```

Comment:
Article usage
`right node is an union of three cylinders` => `right node is a union of three cylinders`

NOTE - since usage of a/an is driven more by the starting sound of the word (`you`nion in union ) rather than the starting vowel (`u` in union).

NOTE from Grammarly - The article an may be incorrect. Consider changing it to agree with the beginning sound of the following word `union`.